test_programs/compile_success_empty/enums/src/main.nr

```diff
@@ -3,9 +3,10 @@ fn main() {
     let _b: Foo<u16> = Foo::B(3);
     let _c = Foo::C(4);
 
-    // (#7172): Single variant enums must be called as functions currently
     let _d: fn() -> Foo<(i32, i32)> = Foo::D;
```

Comment:
This will work when the variant is `D()`?

Reply:
Yes, that should call D as a zero argument function in this case